src/legacy/core_plugins/data/public/search/search_bar/components/saved_query_manager.tsx

```diff
@@ -0,0 +1,360 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPagination,
+  EuiText,
+  EuiConfirmModal,
+  EuiOverlayMask,
+  EuiScreenReaderOnly,
+  EuiIconTip,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import React, { FunctionComponent, useEffect, useState, Fragment } from 'react';
+import { sortBy } from 'lodash';
+import { SavedQuery } from '../index';
+import { getAllSavedQueries, deleteSavedQuery } from '../lib/saved_query_service';
+import { Query } from '../../../query';
+
+const pageCount = 50;
+
+interface Props {
+  showSaveQuery?: boolean;
+  loadedSavedQuery?: SavedQuery;
+  onSave: () => void;
+  onSaveAsNew: () => void;
+  onLoad: (savedQuery: SavedQuery) => void;
+  query: Query;
+  onClearSavedQuery: () => void;
+}
+
+export const SavedQueryManager: FunctionComponent<Props> = ({
```

Comment:
// brain pick - this is a harder naming \ architecture question

The filter service has `FilterMananger` and `FilterBar`, where the former is a service that can be used independently from the UI component.

I wonder if we could stick to this terminology across (data plugin?) services.
Would it be a hassle to rename it to `SavedQueryPopup` \ `SavedQueryListView` or any other name you may find suitable? And then rename `savedQueryService` to `savedQueryManager`?

Let me know what you think.

Reply:
The way I look at it, the FilterManager is a "manager" because it has state and maintains a set of filters. The savedQueryService is a stateless service which is no more than a simple wrapper around the saved objects client. So it would be strange to me to call the savedQueryService a manager. 

We could change the name of the SavedQueryManager component if there is a better one, but we intentionally avoided words like "Popup" or "Popover" in case the UI changes.  ListView is not really descriptive enough since you can do much more than view the queries in the popover. IMO the overlap with FilterManager isn't really confusing since one is a React component and one is a JS class.

Reply:
My best suggestion would be `SavedQueryEditor`.
It's just a hunch, that we might change the stateless implementation into a statefull one.

SO, nice to have, but not critical.

Reply:
We changed `SavedQueryManager` to `SavedQueryManagementComponent` to make it obvious this is a UI component: f68dcc7

> It's just a hunch, that we might change the stateless implementation into a statefull one.

I hope not :)  if we need something akin to the FilterManager for saved queries I think it should probably wrap the service rather than replace it. But filters behave quite differently than saved queries so I don't really see us needing something like that